toctoc/core/src/main/scala/io.buildo.toctoc/core/authentication/AuthenticationDomain.scala

```diff
@@ -5,15 +5,15 @@ package authentication
 import cats.data.EitherT
 import cats.Monad
 
-trait AuthenticationDomain[F[_], C <: Credential] {
+trait AuthenticationDomain[F[_], C] {
```

Comment:
Does it make sense to you to leave `Credential` instead of `C`? Just to make the concept more explicit to the user.
```suggestion
trait AuthenticationDomain[F[_], Credential] {
```